Source/WebCore/rendering/RenderLayerBacking.cpp

```diff
@@ -3185,9 +3185,11 @@ void RenderLayerBacking::updateImageContents(PaintedContentsInfo& contentsInfo)
     if (!image)
         return;
 
+#if (!PLATFORM(GTK) || !PLATFORM(WPE) || !PLATFORM(WIN) || !PLATFORM(PLAYSTATION))
```

Comment:
Wouldn’t PLATFORM(COCOA) be more appropriate here? Also you’re using || instead of &&

Reply:
Good shout on && but I want to avoid (COCOCA) because it might be effecting more platforms like MACCATALYST etc., so I opted to do this instead of macOS restriction.

Reply:
@donny-dont - Fixed '&&' bit in next [commit](https://github.com/WebKit/WebKit/commit/b47cafd505a96dcb0614989074d16d7e2038ebf6).

Reply:
Was thinking more on this and is there a reason why ports wouldn’t want this code enabled?

Reply:
In case of Apple platforms, apple requires all images which go through compositor layer to not load immediately but rendered in one go, which defeat purpose of 'progressive' loading for JPEG (if they have filter or transforms applied or something which them to be composited). So for other ports, this will also delay loading, so I am adding this part behind platform flags to not delay loading if they are composited images on other platform and continue to load progressively.

Reply:
Try loading this https://jsfiddle.net/6y4zjb17/ on Safari vs Chrome.

Reply:
`#if (COCOCA)` seems better to me too.

> it might be effecting more platforms like MACCATALYST etc.

It should effect. It's the original behavior. We should keep Cocoa's behavior.

Reply:
@fujii & @donny-dont - I will wait for @smfr's input to help me understand on which platform, we need to add this flag (e.g., COCOA only or some others as well).

Reply:
Maybe an `ENABLE` flag would be more appropriate here then?